src/main/java/budgetbuddy/logic/rules/performable/SetCategoryExpression.java

```diff
@@ -0,0 +1,30 @@
+package budgetbuddy.logic.rules.performable;
+
+import static budgetbuddy.commons.util.CollectionUtil.requireAllNonNull;
+
+import budgetbuddy.model.Model;
+import budgetbuddy.model.attributes.Category;
+import budgetbuddy.model.rule.expression.Value;
+import budgetbuddy.model.transaction.Transaction;
+
+/**
+ * Represents a set category expression.
+ */
+public class SetCategoryExpression extends PerformableExpression {
+
+    /**
+     * Constructs a SetCategoryExpression with the given value.
+     *
+     * @param value the value to perform the action with.
+     */
+    public SetCategoryExpression(Value value) {
+        super(value);
+    }
+
+    @Override
+    public void perform(Model model, Transaction txn) {
+        requireAllNonNull(model, txn);
+        Category toSet = new Category(value.toString());
+        System.out.println("ADDED CATEGORY!!! " + toSet.toString());
```

Comment:
Might want to add a `todo` here. But up to you, so long as you remember.

Reply:
Ah thanks for reminder